### PR TITLE
Fix printing of "null" values

### PR DESCRIPTION
**Pull request by AviAvni**
See https://github.com/fsharp/fslang-suggestions/issues/364

I'm working with https://github.com/nelak on this as part of mentorship program
please review

image

### CODE REVIEW

**Comment by forki** [comment]
Linking the correct issue: https://github.com/fsharp/fslang-suggestions/issues/364

**Comment by forki** [comment]
you guys should add some tests ;-)

**Comment by AviAvni** [comment]
yeah we will work on it therefore its WIP thenks @forki

**Comment by dsyme** [comment]
awesome

**Comment by AviAvni** [comment]
thanks @dsyme  this also fix printing () as null

File: src/utils/sformat.fs [hunk 1]
```diff
@@ -323,7 +323,15 @@ namespace Microsoft.FSharp.Text.StructuredFormat
                 // to 7.
 
                 if FSharpType.IsTuple reprty then 
-                    TupleValue (FSharpValue.GetTupleFields obj |> Array.toList)
+                    let tyArgs = reprty.GetGenericArguments()
```
Comment (dsyme) [inline src/utils/sformat.fs]: Why not use ``Reflection.FSharpTYpe.GetTupleElements()``?

**Comment by dsyme** [comment]
@AviAvni Please check behaviour with the ``UseNullAsTrueValue`` feature https://msdn.microsoft.com/en-us/visualfsharpdocs/conceptual/core.compilationrepresentationflags-enumeration-%5Bfsharp%5D, and add tests for his case (unless I missed it)

File: src/utils/sformat.fs (hunk 1 above)
Comment (AviAvni) [inline src/utils/sformat.fs]: @dsyme  I'm getting error when using it
image

**Comment by AviAvni** [comment]
@dsyme  the option type also use this attribute so when I'll change the logic to check this also option will work

File: src/utils/sformat.fs (hunk 1 above)
Comment (AviAvni) [inline src/utils/sformat.fs]: using it without the Reflection namespace solvie this

**Comment by AviAvni** [comment]
@dsyme I changed the logic like you asked
Gaston work on the case of RecordValue https://github.com/Microsoft/visualfsharp/pull/1776/files#diff-d394a7d9ed7a227da0c33ec249596510R301 
and he will add more tests

File: src/utils/sformat.fs [hunk 2]
```diff
@@ -368,20 +369,23 @@ namespace Microsoft.FSharp.Text.StructuredFormat
...
-                   if isOptionTy typ then  ConstructorValue("None", [])
-                   elif isUnitType typ then  UnitValue
+                | null ->
+                   let isNullaryUnion =
+                      match typ.GetCustomAttributes(typeof<CompilationRepresentationAttribute>, false) with
+                      | [|:? CompilationRepresentationAttribute as attr|] -> 
+                          (attr.Flags &&& CompilationRepresentationFlags.UseNullAsTrueValue) = CompilationRepresentationFlags.UseNullAsTrueValue
+                      | _ -> false
+                   if isNullaryUnion then
+                     let nullaryCase = FSharpType.GetUnionCases typ |> Array.item 0
```
Comment (AviAvni) [inline src/utils/sformat.fs]: I change it to find the union case with 0 fields

**Comment by dsyme** [comment]
The CI failure is just AppVeyor, probably a timeout.

**Comment by dsyme** [comment]
@KevinRansom I'm happy with this.

**Comment by AviAvni** [comment]
@dsyme all green now

**Comment by dsyme** [comment]
Thank you for this contribution, it's great to have this fixed after all these years

**Pull request merged**